plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer.ts

```diff
@@ -422,7 +457,23 @@ export class SessionRecordingIngester {
                                 gaugeLag.set({ partition }, Math.max(0, metrics.offsetLag))
                             }
 
-                            const recordingMessage = await this.parseKafkaMessage(message, (token) =>
+                            const { token, teamIdWithConfig } = await readTokenFromHeaders(headers, this.teamsRefresher)
+
+                            // NB `==` so we're comparing undefined and null
+                            if (teamIdWithConfig == null || teamIdWithConfig.teamId == null) {
+                                eventDroppedCounter
+                                    .labels({
+                                        event_type: 'session_recordings_blob_ingestion',
+                                        drop_cause: 'team_missing_or_disabled',
+                                    })
+                                    .inc()
+
+                                return statusWarn('team_missing_or_disabled', {
+                                    token: token,
+                                })
+                            }
```

Comment:
Could we not do this in `parseKafkaMessage`? Feels like it fits better inside the function